### PR TITLE
fixed issues with plan create page

### PR DESCRIPTION
**Pull request by briri**
addresses plan creation issues on #777 
- separated out shared controller endpoint that was handling ajax template retrieval and form submission
- added error handler to display notice to user when no templates were found
- disable submit button until a template has been returned
- adjusted integration tests to work with new json endpoint

### CODE REVIEW

File: app/controllers/plans_controller.rb [hunk 1]
```diff
@@ -330,6 +332,52 @@ def set_test
   end
 
 
+  # Collect all of the templates available for the org+funder combination
+  # --------------------------------------------------------------------------
+  def template_options()
```
Comment (jollopre) [inline app/controllers/plans_controller.rb]: Have you considered moving this logic to the templates model?

File: lib/assets/javascripts/views/plans/new.js [hunk 2]
```diff
@@ -49,14 +62,14 @@ $(() => {
...
       $('#plan_template_id option').remove();
 
-      // Fetch the available templates fbased on the funder and research org selected
+      // Fetch the available templates based on the funder and research org selected
       const jQueryForm = $('form#new_plan');
       const formElements = jQueryForm.serializeArray();
       $.ajax({
-        method: getMethod(jQueryForm),
-        url: getAction(jQueryForm),
+        method: 'POST',
```
Comment (jollopre) [inline lib/assets/javascripts/views/plans/new.js]: why are you hardcoding the method, could not you get it from the form method?

File: lib/assets/javascripts/views/plans/new.js [hunk 3]
```diff
@@ -1,17 +1,32 @@
...
+    } else {
+      $('#new_plan button[type="submit"]').attr('disabled', true)
+        .attr('data-toggle', 'tooltip').attr('title', NEW_PLAN_DISABLED_TOOLTIP);
+    }
+  };
+
   // AJAX error function for available template search
   const error = () => {
-    // TODO: add error handling
+    renderAlert(NO_TEMPLATE_FOUND_ERROR);
```
Comment (jollopre) [inline lib/assets/javascripts/views/plans/new.js]: Nice to see here the usage of notification  messages through JS too

File: lib/assets/javascripts/views/plans/new.js (hunk 2 above)
Comment (briri) [reply lib/assets/javascripts/views/plans/new.js]: Its not using the form's `action` in this case so I was reluctant to assume that the method would be the same.

File: app/controllers/plans_controller.rb (hunk 1 above)
Comment (briri) [reply app/controllers/plans_controller.rb]: yes, that would be more appropriate

**Comment by briri** [comment]
- moved template_options endpoint from the plans controller to template controller
- switched to the more appropriate GET method instead of POST
- also rebased with latest from CDL-MVP

**Comment by jollopre** [comment]
Please, fix the merge conflicts and merge back yourself

**Pull request merged**